Exiled.Events/Patches/Generic/Scp173BeingLooked.cs

```diff
@@ -22,12 +29,141 @@ namespace Exiled.Events.Patches.Generic
+    //                        }
+    //                    }
+    //                    else if (__instance._observingPlayers.Contains(currentPlayerReferenceHub))
+    //                    {
+    //                        __instance._observingPlayers.Remove(currentPlayerReferenceHub);
+    //                    }
+    //                }
+    //            }
+
+    // __instance._isObserved = __instance._observingPlayers.Count > 0 || __instance.StareAtDuration > 0f;
+    //            if (count != __instance._observingPlayers.Count && __instance._blinkCooldownRemaining > 0f)
+    //            {
+    //                GameCore.Console.AddDebugLog("SCP173", string.Format("Adjusting blink cooldown. Initial observers: {0}. ", count) + string.Format("New observers: {0}.", __instance._observingPlayers.Count), MessageImportance.LessImportant, false);
+    //                GameCore.Console.AddDebugLog("SCP173", string.Format("Current blink cooldown: {0}", __instance._blinkCooldownRemaining), MessageImportance.LeastImportant, false);
+    //                __instance._blinkCooldownRemaining = Mathf.Max(0f, __instance._blinkCooldownRemaining + ((__instance._observingPlayers.Count - count) * (__instance.BreakneckSpeedsActive ? 0f : 0f)));
+    //                GameCore.Console.AddDebugLog("SCP173", string.Format("New blink cooldown: {0}", __instance._blinkCooldownRemaining), MessageImportance.LeastImportant, false);
+    //                if (__instance._blinkCooldownRemaining <= 0f)
+    //                {
+    //                    __instance.BlinkReady = true;
+    //                }
+    //            }
+    //            HashSetPool<uint>.Shared.Return(hashSet);
+    //            return false;
+    //        }
+    //        catch (Exception generic)
+    //        {
+    //            Log.Error($"Unable to do Scp173BeingLooked {generic}\n");
+    //        }
+
+    // return true;
+    //    }
+    // }
+
     /// <summary>
     /// Patches <see cref="PlayableScps.Scp173.UpdateObservers"/>.
     /// </summary>
     [HarmonyPatch(typeof(PlayableScps.Scp173), nameof(PlayableScps.Scp173.UpdateObservers))]
     internal static class Scp173BeingLooked
     {
+        /// <summary>
+        /// Checks if the current player is to be skipped.
+        /// </summary>
+        /// <param name="instance"> Scp173 instance <see cref="PlayableScps.Scp173"/>. </param>
+        /// <param name="curPlayerGameObj"> Current player game object. </param>
+        /// <param name="curPlayerHub"> Current player referencehub. </param>
+        /// <returns> True if to be skipped, false if not. </returns>
+        public static bool SkipPlayer(ref PlayableScps.Scp173 instance, GameObject curPlayerGameObj, ReferenceHub curPlayerHub)
+        {
+            Player player = Player.Get(curPlayerGameObj);
+            if (player is not null)
+            {
+                if (player?.Role?.Type == RoleType.Tutorial)
```

Comment:
You check `player` for `null` in line above, so `player?.` is not a valid construction(also, as I remember, `player.Role` can't be `null` too)

Reply:
Fair enough, optimization is true, it was a copy paste where I was trying to protect against everything but yes, I will change, good catch